geforcenow.com: [macOS & iPadOS] The wrong values for are reported for the `MouseEvent.button`& `MouseEvent.which` property for extra mouse buttons.

lilyspiniolas opened the pull request:
#### 0c57767cb6c09b63f3158d5eade58b9e2e478440
<pre>
geforcenow.com: [macOS &amp; iPadOS] The wrong values for are reported for the `MouseEvent.button`&amp; `MouseEvent.which` property for extra mouse buttons.
<a href="https://bugs.webkit.org/show_bug.cgi?id=280755">https://bugs.webkit.org/show_bug.cgi?id=280755</a>
<a href="https://rdar.apple.com/137592588">rdar://137592588</a>

Reviewed by Megan Gardner and Aditya Keerthi.

Re-landing 5aee38f with an added build fix.

`MouseEvent.button` and `MouseEvent.which` now return the correct values for
mouse events occuring as a result of a fourth or fifth mouse button being
pressed. Thanks to Abrar Protyasha for starting this patch, and for the work
in 289800@main which is re-landed here as the foundation for these changes.

`MouseEvent.button` and `MouseEvent.which` now return the correct value when
the wheel button is clicked on iOS (macOS already supported this button).

Tested by `fast/events/mouse-event-buttons.html` on non-iOS platforms, and
tested on iOS by new API test `iOSMouseSupport.NonPrimaryMouseButtonsForEvents`.
Added an extra, unrelated test
`iOSMouseSupport.LeftClickFiresPointerAndMouseEvents` to increase test
comprehensiveness for iOS mouse support.

Link WebKitTestRunner against JavaScriptCore on iOS and visionOS to resolve
a build failure that occurs for debug builds.

Tests: fast/events/mouse-event-buttons.html
       Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm

* LayoutTests/fast/events/fire-mousedown-while-pressing-mouse-button.html:
* LayoutTests/fast/events/mouse-click-events-expected.txt:
* LayoutTests/fast/events/mouse-click-events.html:
* LayoutTests/fast/events/mouse-event-buttons-expected.txt: Added.
* LayoutTests/fast/events/mouse-event-buttons.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_mouse-right-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_mouse-right-nonstandard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_pen-right-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_pen-right-nonstandard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_click_on_chorded_mouse_button.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/uievents/mouse/synthetic-mouse-enter-leave-over-out-button-state-after-target-removed.tentative_buttonType=MIDDLE&amp;button=1&amp;buttons=4-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/fast/events/mouse-event-buttons-expected.txt: Added.
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/uievents/mouse/synthetic-mouse-enter-leave-over-out-button-state-after-target-removed.tentative_buttonType=MIDDLE&amp;button=1&amp;buttons=4-expected.txt:
* LayoutTests/platform/win/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:
* Source/WebCore/dom/MouseEvent.cpp:
(WebCore::MouseEvent::button const):
* Source/WebCore/dom/MouseEventTypes.h:
(WebCore::buttonFromShort):
(WebCore::nsEventButtonNumberFromWebCoreMouseButton):
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::currentMouseButton):
(WebCore::buttonFromButtonNumber):
(WebCore::mouseButtonForEvent):
* Source/WebKit/Shared/API/c/WKEvent.h:
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
(WebKit::toAPI):
* Source/WebKit/Shared/WebEvent.serialization.in:
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::platform):
(WebKit::kit):
* Source/WebKit/Shared/WebMouseEvent.h:
* Source/WebKit/Shared/ios/AdditionalButtonMasksIOS.h: Added.
* Source/WebKit/Shared/ios/WebIOSEventFactory.mm:
(WebKit::WebIOSEventFactory::toUIEventButtonMask):
* Source/WebKit/Shared/mac/WebEventFactory.mm:
(WebKit::WebEventFactory::toNSButtonNumber):
* Source/WebKit/UIProcess/API/glib/WebKitPrivate.cpp:
(toWebKitMouseButton):
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(MotionEvent::setState):
(webkitWebViewBaseSynthesizeMouseEvent):
* Source/WebKit/UIProcess/Automation/Automation.json:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::WebAutomationSession):
* Source/WebKit/UIProcess/Automation/gtk/WebAutomationSessionGtk.cpp:
(WebKit::mouseButtonToGdkButton):
* Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionLibWPE.cpp:
(WebKit::mouseButtonToWPEButton):
(WebKit::stateModifierForWPEButton):
* Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp:
(WebKit::libWPEMouseButtonToWPEButton):
* Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm:
(WebKit::WebAutomationSession::sendSynthesizedEventsToPage):
(WebKit::WebAutomationSession::platformSimulateMouseInteraction):
* Source/WebKit/UIProcess/ios/WKMouseInteraction.mm:
(-[WKMouseInteraction createMouseEventWithType:wasCancelled:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(runTest):
* Tools/DumpRenderTree/mac/EventSendingController.h:
* Tools/DumpRenderTree/mac/EventSendingController.mm:
(eventTypeForMouseButtonAndAction):
(swizzledEventPressedMouseButtons):
(swizzledEventButtonNumber):
(-[EventSendingController mouseDown:withModifiers:]):
(-[EventSendingController mouseUp:withModifiers:]):
(-[EventSendingController contextClick]):
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IPCTestingAPI.mm:
(SerializedTypeInfo)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
(-[TestWKWebView pointerInfoAtLocation:]):
((iOSMouseSupport, TrackButtonMaskFromTouchStart)):
((iOSMouseSupport, MouseTimestampTimebase)):
((iOSMouseSupport, EndedTouchesTriggerClick)):
((iOSMouseSupport, CancelledTouchesDoNotTriggerClick)):
((iOSMouseSupport, MouseDidMoveOverElement)):
((iOSMouseSupport, SelectionUpdatesBeforeContextMenuAppears)):
((iOSMouseSupport, DisablingTextIteractionPreventsSelectionWhenShowingContextMenu)):
((iOSMouseSupport, ShowingContextMenuSelectsEditableText)):
((iOSMouseSupport, ShowingContextMenuSelectsNonEditableText)):
((iOSMouseSupport, ContextClickAtEndOfSelection)):
((iOSMouseSupport, WebsiteMouseEventPolicies)):
((iOSMouseSupport, MouseInitiallyDisconnected)):
((iOSMouseSupport, MouseInitiallyConnected)):
((iOSMouseSupport, MouseLaterDisconnected)):
((iOSMouseSupport, MouseLaterConnected)):
((iOSMouseSupport, MouseAlwaysConnected)):
((iOSMouseSupport, BasicPointerInteractionRegions)):
((iOSMouseSupport, LeftClickFiresPointerAndMouseEvents)):
((iOSMouseSupport, NonPrimaryMouseButtonsForEvents)):
* Tools/TestWebKitAPI/Tests/mac/MenuTypesForMouseEvents.mm:
(TestWebKitAPI::buildAndPerformTest):
* Tools/TestWebKitAPI/mac/PlatformWebViewMac.mm:
(TestWebKitAPI::eventTypeForButton):
(TestWebKitAPI::PlatformWebView::simulateButtonClick):
* Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig:
* Tools/WebKitTestRunner/EventSenderProxy.h:
(WTR::EventSenderProxy::lastButtonDown const):
(WTR::EventSenderProxy::mouseButtonsCurrentlyDown const): Deleted.
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/cocoa/EventSenderProxyCocoa.mm: Copied from Source/WebKit/Shared/API/c/WKEvent.h.
(WTR::EventSenderProxy::mouseButtonsCurrentlyDown const):
* Tools/WebKitTestRunner/gtk/EventSenderProxyGtk.cpp:
(WTR::eventSenderButtonToGDKButton):
(WTR::stateModifierForGdkButton):
* Tools/WebKitTestRunner/mac/EventSenderProxy.mm:
(WTR::eventTypeForMouseButtonAndAction):
(WTR::EventSenderProxy::EventSenderProxy):
(WTR::swizzledEventButtonNumber):
(WTR::EventSenderProxy::mouseDown):
(WTR::EventSenderProxy::mouseUp):
(WTR::EventSenderProxy::sendMouseDownToStartPressureEvents):
(): Deleted.

Canonical link: <a href="https://commits.webkit.org/304761@main">https://commits.webkit.org/304761@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/95372d92defadc5a8db3f6d7a7c87e3f9a13e98a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/136421 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/8778 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/47701 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/144133 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/89392 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/85c59eff-172e-458b-9ebb-2e362198014c) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/138293 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/9462 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/8622 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/104320 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/89392 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/caf9addb-87af-4bc3-b701-54a13e884ec2) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/139366 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/6900 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/122233 "Found 1 new API test failure: TestWebKitAPI.InWindowFullscreen.IFrameDocument (failure)") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85156 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/ddc5bacf-d5ab-49f3-bbc3-fb579c2a30de) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/6544 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/4202 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/4725 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/115842 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40439 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/146880 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8460 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41008 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/112655 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/8477 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/7107 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113001 "Found 1 new API test failure: TestWTF:WTF.DragonBox (failure)") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6479 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118543 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/62447 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/21039 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/8508 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/36595 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/8226 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/72067 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/8448 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/8300 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig:
```diff
@@ -31,6 +31,8 @@ PRODUCT_NAME = WebKitTestRunner;
 GCC_ENABLE_OBJC_EXCEPTIONS = YES;
 OTHER_LDFLAGS[sdk=macosx*] = $(inherited) -lWebKitTestRunner -framework Carbon -framework Cocoa -framework JavaScriptCore -framework UniformTypeIdentifiers -framework WebKit;
+OTHER_LDFLAGS[sdk=iphone*] = $(inherited) -framework JavaScriptCore;
+OTHER_LDFLAGS[sdk=xr*] = $(inherited) -framework JavaScriptCore;
```
Shouldn't we also do this for watchOS and tvOS to avoid breaking local builds?

emw-apple replied:
Maybe we should just link it unconditionally?

lilyspiniolas replied:
It is not strictly necessary for watchOS and tvOS debug builds. Should I add anyways?

pxlcoder replied:
Wasn't the issue with the new HashMap? Which I see is under `PLATFORM(COCOA)`.

pxlcoder replied:
Even if it's truly not necessary, I think unconditional linkage would be okay.

lilyspiniolas replied:
It was with the new HashMap, but it really does build for both watchOS and tvOS debug locally in the current state. I'll go ahead and make it unconditional

pxlcoder commented on Source/WebCore/dom/MouseEventTypes.h:
```diff
@@ -42,15 +42,32 @@ enum class SyntheticClickType : uint8_t { NoTap, OneFingerTap, TwoFingerTap };
 
+inline unsigned nsEventButtonNumberFromWebCoreMouseButton(WebCore::MouseButton button)
+{
+    using namespace WebCore;
```
Megan's comment about the style checker made me think, do we even need this? This method is in the WebCore namespace.

pxlcoder commented on Source/WebCore/dom/MouseEventTypes.h:
```diff
@@ -42,15 +42,32 @@ enum class SyntheticClickType : uint8_t { NoTap, OneFingerTap, TwoFingerTap };
     return isKnownButton ? static_cast<MouseButton>(buttonValue) : MouseButton::Other;
 }
 
+inline unsigned nsEventButtonNumberFromWebCoreMouseButton(WebCore::MouseButton button)
```
`WebCore::` here should also be unnecessary?

lilyspiniolas committed:
geforcenow.com: [macOS & iPadOS] The wrong values for are reported for the `MouseEvent.button`& `MouseEvent.which` property for extra mouse buttons.
https://bugs.webkit.org/show_bug.cgi?id=280755
rdar://137592588

Reviewed by Megan Gardner and Aditya Keerthi.

Re-landing 5aee38f with an added build fix.

`MouseEvent.button` and `MouseEvent.which` now return the correct values for
mouse events occuring as a result of a fourth or fifth mouse button being
pressed. Thanks to Abrar Protyasha for starting this patch, and for the work
in 289800@main which is re-landed here as the foundation for these changes.

`MouseEvent.button` and `MouseEvent.which` now return the correct value when
the wheel button is clicked on iOS (macOS already supported this button).

Tested by `fast/events/mouse-event-buttons.html` on non-iOS platforms, and
tested on iOS by new API test `iOSMouseSupport.NonPrimaryMouseButtonsForEvents`.
Added an extra, unrelated test
`iOSMouseSupport.LeftClickFiresPointerAndMouseEvents` to increase test
comprehensiveness for iOS mouse support.

Link WebKitTestRunner against JavaScriptCore on iOS and visionOS to resolve
a build failure that occurs for debug builds.

Tests: fast/events/mouse-event-buttons.html
       Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm

* LayoutTests/fast/events/fire-mousedown-while-pressing-mouse-button.html:
* LayoutTests/fast/events/mouse-click-events-expected.txt:
* LayoutTests/fast/events/mouse-click-events.html:
* LayoutTests/fast/events/mouse-event-buttons-expected.txt: Added.
* LayoutTests/fast/events/mouse-event-buttons.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_mouse-right-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_mouse-right-nonstandard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_pen-right-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_pen-right-nonstandard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_click_on_chorded_mouse_button.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/uievents/mouse/synthetic-mouse-enter-leave-over-out-button-state-after-target-removed.tentative_buttonType=MIDDLE&button=1&buttons=4-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/fast/events/mouse-event-buttons-expected.txt: Added.
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/uievents/mouse/synthetic-mouse-enter-leave-over-out-button-state-after-target-removed.tentative_buttonType=MIDDLE&button=1&buttons=4-expected.txt:
* LayoutTests/platform/win/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:
* Source/WebCore/dom/MouseEvent.cpp:
(WebCore::MouseEvent::button const):
* Source/WebCore/dom/MouseEventTypes.h:
(WebCore::buttonFromShort):
(WebCore::nsEventButtonNumberFromWebCoreMouseButton):
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::currentMouseButton):
(WebCore::buttonFromButtonNumber):
(WebCore::mouseButtonForEvent):
* Source/WebKit/Shared/API/c/WKEvent.h:
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
(WebKit::toAPI):
* Source/WebKit/Shared/WebEvent.serialization.in:
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::platform):
(WebKit::kit):
* Source/WebKit/Shared/WebMouseEvent.h:
* Source/WebKit/Shared/ios/AdditionalButtonMasksIOS.h: Added.
* Source/WebKit/Shared/ios/WebIOSEventFactory.mm:
(WebKit::WebIOSEventFactory::toUIEventButtonMask):
* Source/WebKit/Shared/mac/WebEventFactory.mm:
(WebKit::WebEventFactory::toNSButtonNumber):
* Source/WebKit/UIProcess/API/glib/WebKitPrivate.cpp:
(toWebKitMouseButton):
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(MotionEvent::setState):
(webkitWebViewBaseSynthesizeMouseEvent):
* Source/WebKit/UIProcess/Automation/Automation.json:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::WebAutomationSession):
* Source/WebKit/UIProcess/Automation/gtk/WebAutomationSessionGtk.cpp:
(WebKit::mouseButtonToGdkButton):
* Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionLibWPE.cpp:
(WebKit::mouseButtonToWPEButton):
(WebKit::stateModifierForWPEButton):
* Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp:
(WebKit::libWPEMouseButtonToWPEButton):
* Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm:
(WebKit::WebAutomationSession::sendSynthesizedEventsToPage):
(WebKit::WebAutomationSession::platformSimulateMouseInteraction):
* Source/WebKit/UIProcess/ios/WKMouseInteraction.mm:
(-[WKMouseInteraction createMouseEventWithType:wasCancelled:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(runTest):
* Tools/DumpRenderTree/mac/EventSendingController.h:
* Tools/DumpRenderTree/mac/EventSendingController.mm:
(eventTypeForMouseButtonAndAction):
(swizzledEventPressedMouseButtons):
(swizzledEventButtonNumber):
(-[EventSendingController mouseDown:withModifiers:]):
(-[EventSendingController mouseUp:withModifiers:]):
(-[EventSendingController contextClick]):
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IPCTestingAPI.mm:
(SerializedTypeInfo)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
(-[TestWKWebView pointerInfoAtLocation:]):
((iOSMouseSupport, TrackButtonMaskFromTouchStart)):
((iOSMouseSupport, MouseTimestampTimebase)):
((iOSMouseSupport, EndedTouchesTriggerClick)):
((iOSMouseSupport, CancelledTouchesDoNotTriggerClick)):
((iOSMouseSupport, MouseDidMoveOverElement)):
((iOSMouseSupport, SelectionUpdatesBeforeContextMenuAppears)):
((iOSMouseSupport, DisablingTextIteractionPreventsSelectionWhenShowingContextMenu)):
((iOSMouseSupport, ShowingContextMenuSelectsEditableText)):
((iOSMouseSupport, ShowingContextMenuSelectsNonEditableText)):
((iOSMouseSupport, ContextClickAtEndOfSelection)):
((iOSMouseSupport, WebsiteMouseEventPolicies)):
((iOSMouseSupport, MouseInitiallyDisconnected)):
((iOSMouseSupport, MouseInitiallyConnected)):
((iOSMouseSupport, MouseLaterDisconnected)):
((iOSMouseSupport, MouseLaterConnected)):
((iOSMouseSupport, MouseAlwaysConnected)):
((iOSMouseSupport, BasicPointerInteractionRegions)):
((iOSMouseSupport, LeftClickFiresPointerAndMouseEvents)):
((iOSMouseSupport, NonPrimaryMouseButtonsForEvents)):
* Tools/TestWebKitAPI/Tests/mac/MenuTypesForMouseEvents.mm:
(TestWebKitAPI::buildAndPerformTest):
* Tools/TestWebKitAPI/mac/PlatformWebViewMac.mm:
(TestWebKitAPI::eventTypeForButton):
(TestWebKitAPI::PlatformWebView::simulateButtonClick):
* Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig:
* Tools/WebKitTestRunner/EventSenderProxy.h:
(WTR::EventSenderProxy::lastButtonDown const):
(WTR::EventSenderProxy::mouseButtonsCurrentlyDown const): Deleted.
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/cocoa/EventSenderProxyCocoa.mm: Copied from Source/WebKit/Shared/API/c/WKEvent.h.
(WTR::EventSenderProxy::mouseButtonsCurrentlyDown const):
* Tools/WebKitTestRunner/gtk/EventSenderProxyGtk.cpp:
(WTR::eventSenderButtonToGDKButton):
(WTR::stateModifierForGdkButton):
* Tools/WebKitTestRunner/mac/EventSenderProxy.mm:
(WTR::eventTypeForMouseButtonAndAction):
(WTR::EventSenderProxy::EventSenderProxy):
(WTR::swizzledEventButtonNumber):
(WTR::EventSenderProxy::mouseDown):
(WTR::EventSenderProxy::mouseUp):
(WTR::EventSenderProxy::sendMouseDownToStartPressureEvents):
(): Deleted.

Canonical link: https://commits.webkit.org/304761@main
